### PR TITLE
Support for Conway ref-script fees

### PR DESCRIPTION
**Pull request by groscoe**
Unfortunately but understandably, Lucid's vendored version of CML lags the upstream by a lot, so I'm making the changes here instead. The important changes are in `tx_builder.rs`.

Resolves SMA-493.

### CODE REVIEW

File: src/core/libs/cardano_multiplatform_lib/src/tx_builder.rs [hunk 1]
```diff
@@ -252,11 +253,46 @@ fn min_fee(tx_builder: &mut TransactionBuilder) -> Result<Coin, JsError> {
...
-    fees::min_fee(
+    let base_fee = fees::min_fee(
         &full_tx,
         &tx_builder.config.fee_algo,
         &tx_builder.config.ex_unit_prices,
-    )
+    ).unwrap();
+
+    let mut reference_scripts_tier_price = BigNum::from(0);
+    if let Some(reference_inputs) = &tx_builder.reference_inputs.clone() {
```
Comment (waalge) [inline src/core/libs/cardano_multiplatform_lib/src/tx_builder.rs]: Include inputs as well as reference inputs? 

https://github.com/IntersectMBO/cardano-ledger/blob/7e65f0365eef647b9415e3fe9b3c35561761a3d5/eras/conway/impl/src/Cardano/Ledger/Conway/UTxO.hs#L155

Comment (waalge) [reply src/core/libs/cardano_multiplatform_lib/src/tx_builder.rs]: Timely: 
https://github.com/spacebudz/lucid/blob/a0e182fe77d553f6480d2b3ff5d1f38eee22e4e9/src/core/libs/cardano_multiplatform_lib/src/tx_builder.rs#L258

Comment (groscoe) [reply src/core/libs/cardano_multiplatform_lib/src/tx_builder.rs]: Thanks!

Comment (waalge) [reply src/core/libs/cardano_multiplatform_lib/src/tx_builder.rs]: My patch
https://github.com/kompact-io/lucid/blob/2d0021dbd0605a53346885a81191e25f1ab3ba7b/src/core/libs/cardano_multiplatform_lib/src/tx_builder.rs#L262
(Could not find the right incantation to output the right wasm/js). 
This seems to be passing.

Comment (groscoe) [reply src/core/libs/cardano_multiplatform_lib/src/tx_builder.rs]: Hey @waalge, I don't have access to your patch, but the upstream implementation you liked above seems to be wrong. It assumes a fixed cost per byte of 15 lovelace, when in fact the cost per byte increases with the total ref script size (the actual formula is `costPerByte = base * multiplier^(floor(costPerByte / range))`, where `base`, `multiplier` and `range` are new Conway-era protocol parameters.

Comment (waalge) [reply src/core/libs/cardano_multiplatform_lib/src/tx_builder.rs]: @groscoe Apologies, link should now be public. 

I tried running of lucids' branches and hit wasm issues. 

~~I think the code is slightly obscured. Dur to the reason commented https://github.com/Liqwid-Labs/lucid/pull/1/files#r1711328007~~
woops. I was looking at your code thinking it was from lucid. I think you're right

**Review by kylixafonso — APPROVED** [review]
LGTM!

**Pull request merged**